From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: net: dsa: microchip: call phy_remove_link_mode during probe
Patch-mainline: never, kabi
References: kabi

In networking-stable-20_07_29, commit 3506b2f42dff (net: dsa: microchip:
call phy_remove_link_mode during probe) removed one member from struct
ksz_dev_ops. This made the kABI checker complaining.

Reintroduce the member as a placeholder.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 drivers/net/dsa/microchip/ksz_priv.h |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/net/dsa/microchip/ksz_priv.h
+++ b/drivers/net/dsa/microchip/ksz_priv.h
@@ -120,6 +120,8 @@ struct ksz_dev_ops {
 	u32 (*get_port_addr)(int port, int offset);
 	void (*cfg_port_member)(struct ksz_device *dev, int port, u8 member);
 	void (*flush_dyn_mac_table)(struct ksz_device *dev, int port);
+	void (*phy_setup)(struct ksz_device *dev, int port,
+			struct phy_device *phy); /* kABI; unused */
 	void (*port_cleanup)(struct ksz_device *dev, int port);
 	void (*port_setup)(struct ksz_device *dev, int port, bool cpu_port);
 	void (*r_phy)(struct ksz_device *dev, u16 phy, u16 reg, u16 *val);
